core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java

```diff
@@ -60,8 +60,8 @@
     private int mask;
 
     // the hash table from values to groupIds
-    private LongBigArray values;
```

Comment:
Is this based on the understanding that https://bugs.openjdk.java.net/browse/JDK-8027959 has solved the problem of GC pauses due to humongous objects ?
Could we still suffer from OOMs due to heap fragmentation as reported in http://mail.openjdk.java.net/pipermail/hotspot-gc-use/2017-November/002725.html (https://bugs.openjdk.java.net/browse/JDK-8191565) ?
I guess our HeapRegionSize=32M jam config recommendation should mitigate it to some extent.

Reply:
> Could we still suffer from OOMs due to heap fragmentation as reported in http://mail.openjdk.java.net/pipermail/hotspot-gc-use/2017-November/002725.html (https://bugs.openjdk.java.net/browse/JDK-8191565) ?

Possibly, but IMO gains are worth it. Note that `MultiChannelGroupByHash` already uses raw arrays (albeit I suspect it will have less groups on average as multi-channel group by are based on user `GROUP BY`).

I think https://github.com/trinodb/trino/pull/11011 will mitigate risk of fragmenting memory.